admin/handler/bitbucket.go

```
@@ -21,20 +22,24 @@ const BitbucketRepoBaseV1 = "https://api.bitbucket.org/1.0/repositories/%s"
 type Bitbucket struct {
 	Client    ocenet.HttpClient
 	Marshaler jsonpb.Marshaler
+
+	credConfig	*models.AdminConfig
+	isInitialized	bool
 }
 
-// Takes in admin config creds, sets Client and Marshaler Attributes
-func (bb *Bitbucket) SetMeUP(adminConfig *models.AdminConfig) {
-	var conf = clientcredentials.Config{
+//TODO: why is up in all lowercase?
+// Takes in admin config creds, return true if successful, false if fails
+func (bb *Bitbucket) SetMeUP(adminConfig *models.AdminConfig) bool {
```

should this maybe just return the error that would set the bool to false? that way we can bubble that error up and return it if an a bb account is set up via api call

---

I was hoping to do that, but unfortunately the code that actually sets up the creds are inside of a for/range loop over a configCreds channel. The function that handles the REST call just puts a cred into a channel and exits, I don't know if I can make the listener inside of ListenForConfig return an error that would somehow get sent back to the calling function (ConfigHandler). You know what I mean?

I'll look into it though, if both of us are thinking it might be nice to do error reporting for the rest endpoint then probably worth doing some research and possibly reevaluate using channels.

---

can there be an error channel and a config channel?

---

i'll look at this code more

---

oooo error channel and config channel is great idea - ConfigHandler can listen off of error channel and if it gets one, just return it. The only thing I can think of that might kind of suck is if two people are setting configs at the same time, and both errors out, I wonder how we can guarantee that the right error gets sent back to the right caller

---

Looking at code now, I'm thinking of possibly removing config channel altogether?

---

i swear i've seen this somewhere in one of the random tutorials i did lol

---

I think question here is if channels preserve ordering or if they return whichever one comes back first

---

I just realized...for the rest api (and for reading config from file), the walk can't be async. Not being able to create webhooks is an error that we'd want to propagate back to the user, so no matter what they're gonna have to wait. If there's no objections I think I'm gonna change it all to be blocking till Bitbucket's functions return. If you have large repo for walk, you'll just have to wait a bit, I guess? AHHH I DON'T KNOW

---

is the walk going through every file in the repo, or just the first level? because i'm alright with enforcing it being in the root dir, they can set a relative path for the "build" part to run in (ie src)
That at least wont' be waiting as long... idk though lets make it blocking for now, optimize later. so many decisions are so hard without foresight 🔮 professor trelawney where u at